optuna/_callbacks.py

```diff
@@ -64,8 +65,15 @@ class RetryFailedTrialCallback:
     When a trial fails, this callback can be used with the :class:`optuna.storage` class to
     recreate the trial in :obj:`TrialState.WAITING` to queue up the trial to be run again.
 
-    This is helpful in environments where trials may fail due to external conditions, such as
-    being preempted by other processes.
+    The failed trial can be identified by the
+    :func:`~optuna.storages.RetryFailedTrialCallback.retried_trial_number` function.
+    Even if repetitive failure occurrs (a retried trial fails again),
```

Comment:
A nit. Let me commit by myself.
```suggestion
    Even if repetitive failure occurs (a retried trial fails again),
```